Assets/LeapMotion/Scripts/ReporterBase.cs

```diff
@@ -0,0 +1,81 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public abstract class ReporterBase : MonoBehaviour {
+  protected KeyCode m_safetyKey;
+  protected KeyCode m_triggerKey;
+
+  protected enum RecordingState {
+    READY,
+    RECORDING,
+    SAVING,
+    REPLAYING
+  }
+  private RecordingState m_recordingState = RecordingState.READY;
+
+  public void SetKeys(KeyCode safetyKey, KeyCode triggerKey) {
+    m_safetyKey = safetyKey;
+    m_triggerKey = triggerKey;
+  }
+
+  public bool IsReady() { return m_recordingState == RecordingState.READY; }
+  public bool IsRecording() { return m_recordingState == RecordingState.RECORDING; }
+  public bool IsSaving() { return m_recordingState == RecordingState.SAVING; }
+  public bool IsReplaying() { return m_recordingState == RecordingState.REPLAYING; }
+
+  public void TriggerStartRecording() {
+    if (!IsReady())
+      return;
+    if (StartRecording())
```

Comment:
This allows for failure of individual ReporterBase instances, making them out of sync with ReporterManager.
This assumes that a failure still allows for subsequent transition-to-state requests to be handled correctly.